### PR TITLE
feat: support for scheduling of cron jobs in JobPlugin

### PR DESCRIPTION
**Pull request by collinlokken**
## What does this pull request change?
Adds support for cron job in JobPlugin

## Why is this pull request needed?
Makes it possible to schedule the job for later

### CODE REVIEW

File: packages/dm-core/src/types.ts [hunk 1]
```diff
@@ -100,6 +100,14 @@ export type TJob = {
   ended?: string
   outputTarget?: string
   referenceTarget?: string
+  schedule?: TSchedule
+}
+
+export type TSchedule = {
+  type: string
```
Comment (henrikbossart) [inline packages/dm-core/src/types.ts:107]: Are we planning on having different kinds of schedules? 🤔

**Review by henrikbossart — APPROVED** [review]

**Review by soofstad — CHANGES_REQUESTED** [review]
Instead of going multiple rounds on this.
I suggest we bring in the old "cronJob"-code and do it properly the first time.
https://github.com/equinor/dm-app-analysis-platform/blob/main/src/plugins/analysis-platform/modules/Jobs/not_used/ReoccurringJob.tsx
There is no big rush getting this into the frontend yet

File: packages/dm-core-plugins/src/job/DateRangePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { TextField } from '@equinor/eds-core-react'
+
+const DateRangePicker = (props: {
+  setDateRange: (dateRange: { startDate: string; endDate: string }) => void
+  value: any
+}): JSX.Element => {
+  const { setDateRange, value } = props
+
+  console.log(value)
```
Comment (eoaksnes) [inline packages/dm-core-plugins/src/job/DateRangePicker.tsx:10]: Please remove console.log

File: packages/dm-core-plugins/src/job/JobPlugin.tsx [hunk 3]
```diff
@@ -117,12 +141,37 @@ export const JobPlugin = (props: IUIPlugin & { config: JobPluginConfig }) => {
...
+        }
+        checked={asCronJob}
+      />
+      {asCronJob && (
+        <CreateReoccurringJob
+          cronJob={jobSchedule}
+          close={() => setAsCronJob(false)}
+          removeJob={() => setJobSchedule(emptyJob)}
+          setCronJob={(e: TSchedule) => {
+            console.log(e)
```
Comment (eoaksnes) [inline packages/dm-core-plugins/src/job/JobPlugin.tsx:163]: Please remove console.log

File: packages/dm-core-plugins/src/job/CronJob.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+          setDateRange={(dateRange) => setDateRange(dateRange)}
+          value={dateRange}
+        />
+        <InputWrapper>
+          <div style={{ display: 'flex', flexDirection: 'column' }}>
+            <Label label="Interval" />
+            <StyledSelect
+              onChange={(e) => {
+                //@ts-ignore
+                // TODO fix type
```
Comment (eoaksnes) [inline packages/dm-core-plugins/src/job/CronJob.tsx:104]: Suggestion: Would be nice if we can remove the ts-ignore.

File: packages/dm-core-plugins/src/job/CronJob.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+          )}
+          {interval === EInterval.HOURLY && (
+            <div style={{ display: 'flex', flexDirection: 'column' }}>
+              <Label label="Hour step" />
+              <StyledSelect
+                onChange={(e) => {
+                  setHourStep(e.target.value)
+                }}
+              >
+                {/*TODO fix type error*/}
```
Comment (eoaksnes) [inline packages/dm-core-plugins/src/job/CronJob.tsx:143]: Suggestion: Would be nice if we can remove the ts-ignore.

**Review by eoaksnes — APPROVED** [review]
Two console.log. and suggest to remove the ts-ignore

File: packages/dm-core-plugins/src/job/JobPlugin.tsx [hunk 6]
```diff
@@ -79,12 +83,7 @@ export const JobPlugin = (props: IUIPlugin & { config: JobPluginConfig }) => {
...
   const [asCronJob, setAsCronJob] = useState<boolean>(false)
-  const [jobSchedule, setJobSchedule] = useState<TSchedule>({
-    type: EBlueprint.CRON_JOB,
-    cron: defaultCron,
-    startDate: '',
-    endDate: '',
-  })
+  const [jobSchedule, setJobSchedule] = useState<TSchedule>(emptyJob)
   const canSubmit =
     !asCronJob || Boolean(jobSchedule.startDate && jobSchedule.endDate)
```
Comment (soofstad) [inline packages/dm-core-plugins/src/job/JobPlugin.tsx:88]: Know this is outside current PR.
But, no need to set this requirement on start- endDate when we don't use them for anything

**Review by soofstad — APPROVED** [review]

File: packages/dm-core-plugins/src/job/JobPlugin.tsx (hunk 6 above)
Comment (collinlokken) [reply packages/dm-core-plugins/src/job/JobPlugin.tsx:88]: We can disable the check, but I suggest that we do it in another PR

**Pull request merged**